### PR TITLE
Upgrade @mui/x-data-grid to v6 

### PR DESCRIPTION
**Pull request by nickgros**
- Upgrade @mui/x-data-grid (used only in Challenge Portal components) to v6
- Modify `RadioOption` component to use `checked` instead of `currentValue` (only affects challenge portal components)
- Bump other https://github.com/mui packages to latest version

### CODE REVIEW

File: packages/synapse-react-client/src/components/widgets/RadioGroup.tsx [hunk 1]
```diff
@@ -34,10 +34,10 @@ export function RadioGroup<T extends string | boolean | number = string>(
 export type RadioOptionProps<T extends string | boolean | number = string> = {
   label: React.ReactNode
   value: T
-  currentValue?: T
   style?: React.CSSProperties
   onChange: (value: T) => void
   disabled?: boolean
+  checked: boolean
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/widgets/RadioGroup.tsx:40]: Replace `currentValue` with `checked`, which is a bit more intuitive/more closely matches the `input` interface

File: packages/synapse-react-client/package.json [hunk 2]
```diff
@@ -33,12 +33,12 @@
...
-    "@mui/material": "^5.14.17",
-    "@mui/system": "^5.14.17",
-    "@mui/utils": "^5.14.17",
-    "@mui/x-data-grid": "5.17.26",
-    "@mui/x-date-pickers": "^6.18.1",
+    "@mui/icons-material": "^5.15.10",
+    "@mui/material": "^5.15.10",
+    "@mui/system": "^5.15.9",
+    "@mui/utils": "^5.15.9",
+    "@mui/x-data-grid": "^6.19.4",
```
Comment (nickgros) [inline packages/synapse-react-client/package.json:40]: Upgrade @mui/x-data-grid to v6. Like x-date-pickers below, it's ok to mix v6 with @mui/material v5

This change only affects challenge portal components.

File: packages/synapse-react-client/src/components/ChallengeSubmission/EvaluationQueueList.tsx [hunk 3]
```diff
@@ -84,9 +84,8 @@ function EvaluationQueueList({
         density="compact"
         autoHeight
         loading={isLoading}
-        components={{
-          Header: () => null,
-          Toolbar: () => null,
+        slots={{
+          toolbar: () => null,
         }}
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/ChallengeSubmission/EvaluationQueueList.tsx:89]: [Per migration guide](https://mui.com/x/migration/migration-data-grid-v5/#removed-props):
> The components.Header slot was removed. Use components.Toolbar or slots.toolbar slot instead.

File: packages/synapse-react-client/src/components/ChallengeSubmission/EvaluationQueueList.tsx [hunk 4]
```diff
@@ -39,7 +39,7 @@ function EvaluationQueueList({
         return (
           <RadioOption
             value={params.id as string}
-            currentValue={selectedEvaluation}
+            checked={params.id === selectedEvaluation}
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/ChallengeSubmission/EvaluationQueueList.tsx:42]: Changed our `RadioOption` to be a bit more conventional

File: packages/synapse-react-client/src/components/ChallengeSubmission/SubmissionCommitList.tsx [hunk 5]
```diff
@@ -134,19 +134,17 @@ function SubmissionCommitList({
...
           loading={isLoading}
           columns={columns}
           rows={getPageData()}
-          pageSize={PER_PAGE}
           rowCount={commitResults?.totalNumberOfResults ?? 0}
-          page={page}
           pagination
           paginationMode="server"
-          onPageChange={n => handlePageChange(n)}
+          paginationModel={{ page, pageSize: PER_PAGE }}
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/ChallengeSubmission/SubmissionCommitList.tsx:143]: [Per migration guide](https://mui.com/x/migration/migration-data-grid-v5/#pagination):
> * The page and pageSize props and their respective event handlers onPageChange and onPageSizeChange were removed. Use paginationModel and [onPaginationModelChange](https://mui.com/x/react-data-grid/pagination/#controlled-pagination-model) instead.
> * The properties initialState.pagination.page and initialState.pagination.pageSize were also removed. Use [initialState.pagination.paginationModel](https://mui.com/x/react-data-grid/pagination/#initializing-the-pagination-model) instead.
> * The rowsPerPageOptions prop was renamed to [pageSizeOptions](https://mui.com/x/react-data-grid/pagination/#page-size-options).

File: packages/synapse-react-client/package.json [hunk 6]
```diff
@@ -33,12 +33,12 @@
...
+    "@mui/x-date-pickers": "^6.19.4",
```
Comment (jay-hodgson) [inline packages/synapse-react-client/package.json:41]: Did you try out the updated date picker?

**Review by jay-hodgson — APPROVED** [review]

File: packages/synapse-react-client/package.json (hunk 6 above)
Comment (nickgros) [reply packages/synapse-react-client/package.json:41]: No, but the upgrade here for `@mui/x-date-pickers` was a minor version bump that we could have picked up anyways with the current semver notation (we were already on v6). There should not be any breaking changes.

**Pull request merged**